payments-core/src/main/java/com/stripe/android/networking/StripeApiRepository.kt

```diff
@@ -1458,6 +1458,28 @@ class StripeApiRepository @JvmOverloads internal constructor(
         )
     }
 
+    override suspend fun deletePaymentDetails(
+        clientSecret: String,
+        paymentDetailsId: String,
+        requestOptions: ApiRequest.Options
+    ): Result<Unit> {
+        return makeApiRequest(
+            apiRequestFactory.createDelete(
+                getConsumerPaymentDetailsUrl(paymentDetailsId),
+                requestOptions,
+                mapOf(
+                    "request_surface" to "android_payment_element",
+                    "credentials" to mapOf(
+                        "consumer_session_client_secret" to clientSecret
+                    )
+                )
+            ),
+            onResponse = {}
+        ).runCatching {
```

Comment:
why do we need this runCatching block? If this is to convert exceptions in the api request into Result.failures, can we wrap that call in the runCatching, rather than having it here? I think that's more consistent with what's done elsewhere in this repo